docs/proposals/01-extensibility.md

```diff
@@ -0,0 +1,844 @@
+If both match they know that they should act on the resource, otherwise they stop doing anything.
+
+:information: We will provide an easy-to-use framework for the controllers containing all of these features out-of-the-box in order to allow the developers to focus on writing the actual controller logic.
+
+When a Seed control plane move is triggered, the `.spec.cloud.seed` field of the respective `Shoot` is changed.
+Gardener will change the respective DNS record's value (`aws-01.core.garden.example.com`) to contain the new Seed name.
+After that it will wait `2*60s` to be sure that all controllers have observed the change.
+Then it starts reconciling and applying the CRDs together with a preset `.status.state` into the new Seed (based on its last observations which were stored in the respective `ShootState` object stored in the Garden cluster).
+The controllers are - as per contract - asked to reconstruct their own environment based on the `.status.state` they have written before and the real world's status.
+Apart from that, the normal reconciliation flow gets executed.
+
+Gardener stores the list of Seeds that were responsible for hosting a Shoots control plane at some time in the Shoots `.status.seeds` list so that it knows which Seeds must be cleaned up (i.e., where the control plane must be deleted because it has been moved).
+Once cleaned up, the Seed's name will be removed from that list.
+
+### BackupInfrastructure migration
+
+One part of the reconciliation flow above is the provisioning of the infrastructure for the Shoot's etcd backups (usually, this is a blob store bucket/container).
+Gardener already uses a separate `BackupInfrastructure` resource that is written into the Garden cluster and picked up by a dedicated `BackupInfrastructure` controller (bundled into the Gardener controller manager).
+This dedicated resource exists mainly for the reason to allow keeping backups for a certain "grace period" even after the Shoot deletion itself:
+
+```yaml
+apiVersion: gardener.cloud/v1alpha1
+kind: BackupInfrastructure
+metadata:
+  name: aws-01-bucket
+  namespace: garden-core
+spec:
+  seed: seed-01
+  shootUID: uuid-of-shoot
+```
+
+The actual provisioning is executed in a corresponding Seed cluster as Gardener can only assume network connectivity to the underlying cloud environment in the Seed.
+We would like to keep the created artifacts in the Seed (e.g., Terraform state) near to the control plane.
+Consequently, when Gardener moves a control plane, it will update the `.spec.seed` field of the `BackupInfrastructure` resource as well.
+With the exact same logic described above the `BackupInfrastructure` controller inside the Gardener will move to the new Seed.
+
+## Registration of external controllers at Gardener
+
+We want to have a dynamic registration process, i.e. we don't want to build which controllers shall be deployed inside the Gardener Docker images.
```

Comment:
we don't want to build which controllers shall be deployed: I don't understand this